app/lib/differential_expression_service.rb

```diff
@@ -1,5 +1,116 @@
 # handle launching differential expression ingest jobs
 class DifferentialExpressionService
+  # run a differential expression job for a given study on the default cluster/annotation
+  #
+  # * *params*
+  #   - +study_accession+ (String) => Accession of study to use
+  #   - +user+ (User) => Corresponding user, will default to study owner
+  #
+  # * *yields*
+  #   - (IngestJob) => Differential expression job in PAPI
+  #
+  # * *returns*
+  #   - (Boolean) => True if job queues successfully
+  #
+  # * *raises*
+  #   - (ArgumentError) => if requested parameters do not validate
+  def self.run_differential_expression_on_default(study_accession, user: nil)
+    study = Study.find_by(accession: study_accession)
+    validate_study(study)
+    raise ArgumentError, "#{study.accession} has no default cluster" if study.default_cluster.blank?
+    raise ArgumentError, "#{study.accession} has no default annotation" if study.default_annotation.blank?
+
+    annotation_name, annotation_type, annotation_scope = study.default_annotation.split('--')
+    annotation = {
+      annotation_name: annotation_name,
+      annotation_type: annotation_type,
+      annotation_scope: annotation_scope
+    }
+    cluster_file = study.default_cluster.study_file
+    requested_user = user || study.user
+    run_differential_expression_job(cluster_file, study, requested_user, **annotation)
+  end
+
+  # same as above method, except runs differential expression job on all valid annotations
+  #
+  # * *params*
+  #   - +study_accession+ (String) => Accession of study to use
+  #   - +user+ (User) => Corresponding user, will default to study owner
+  #
+  # * *yields*
+  #   - (IngestJob) => Differential expression job in PAPI for each valid cluster/annotation combination
+  #
+  # * *returns*
+  #   - (Integer) => Number of DE jobs yielded
+  #
+  # * *raises*
+  #   - (ArgumentError) => if requested study cannot run any DE jobs
+  def self.run_differential_expression_on_all(study_accession, user: nil)
+    study = Study.find_by(accession: study_accession)
+    validate_study(study)
+    available_annotations = []
+
+    metadata = study.cell_metadata.where(annotation_type: 'group', is_differential_expression_enabled: false)
+                    .select(&:can_visualize?)
+    available_annotations += metadata.map do |meta|
```

Comment:
I like Jean's term "eligible" here, as I think it better connotes that we're applying more selection criteria for these annotations.  Ideally we'd change how we talk of these things or change the code to align with how we talk.

```suggestion
    eligible_annotations += metadata.map do |meta|
```